app/Policies/Admin/UserPolicy.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+namespace App\Policies\Admin;
+
+use App\User;
+use Illuminate\Support\Arr;
+use Illuminate\Auth\Access\HandlesAuthorization;
+
+class UserPolicy
+{
+    use HandlesAuthorization;
+
+    /**
+     * Determine whether the user can view any models.
+     *
+     * @param User $user
+     * @return mixed
+     */
+    public function viewAny(User $user)
+    {
+        return $user->hasAnySpecifiedPermsOnAnyWiki(['tooladmin']);
+    }
+
+    /**
+     * Determine whether the user can view the model.
+     *
+     * @param User $user
+     * @param User $model
+     * @return mixed
+     */
+    public function view(User $user, User $model)
+    {
+        if ($user->id === $model->id) {
+            // allow user to view their own profile
+            return true;
+        }
+
+        return $user->hasAnySpecifiedPermsOnAnyWiki(['tooladmin']);
+    }
+
+    /**
+     * Determine whether the user can create models.
+     *
+     * @param User $user
+     * @return mixed
+     */
+    public function create(User $user)
+    {
+        return $user->hasAnySpecifiedPermsOnAnyWiki(['tooladmin']);
+    }
```

Comment:
Even though I don't see the implementation of this, no one should be able to create a user as it's meant to be OAuth managed